File: Easy/404-Sum of Left Leaves/404.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        
        return DFS(root, false);
    }
    
    int DFS(TreeNode *root, bool isLeft)
    {
        if(!root)
            return 0;
        if(!root->left && !root->right)
            return isLeft ? root->val : 0;

        return DFS(root->left, true) + DFS(root->right, false);
    }
};

File: Easy/404-Sum of Left Leaves/404.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.DFS(root, False)
    def DFS(self, root, isLeft):
        
        
        if(not root):
            return 0
        if(not root.left and not root.right):
            return root.val if isLeft else 0
        return self.DFS(root.left, True) + self.DFS(root.right, False)